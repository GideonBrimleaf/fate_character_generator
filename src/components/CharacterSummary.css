.characterSummaryTile {
  background-color: lightcyan;
  text-decoration: none;
  margin: 1rem;
  flex-basis: 0;
  flex-wrap: wrap;
  min-width: 380px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 1rem
}

.character-summary-details {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.character-skills-table {
  margin-top: auto;
}

.character-skills-table td {
  border-bottom: 1px solid darkgrey;
}

.option-links {
  /* align-self: flex-end; */
  margin-top: auto;
}

.character-link {
  text-decoration: none;
  color: black;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media (max-width:600px) {
  .characterSummaryTile {
    flex-direction: column;
    min-width: 80vw;
    margin: 0rem, 1rem;
  }

  .option-links {
    align-self: auto;
  }
}