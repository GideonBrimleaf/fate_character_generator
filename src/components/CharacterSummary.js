/* eslint no-alert: "off" */
import React from 'react'
import './CharacterSummary.css'
import '../App.css'
import { Link } from 'react-router-dom'
import chunkos from '../lib/chunkos'

const CharacterSummary = (props) => {
  if (!props.characterDetails) return null

  const handleCharacterDeletion = () => {
    if (window.confirm('Are you sure you want to delete this character?')) {
      props.deleteCharacter(props.characterDetails)
    }
  }

  const characterApproaches = Object.keys(props.characterDetails.approaches)

  const skills = chunkos(characterApproaches, 2).reduce((tableRow, skillsPair) => {
    const items = skillsPair.map((skill) => {
      const approach = props.characterDetails.approaches[skill]
      return (
        <td key={props.characterDetails.id + skill} className="summary-skill-header">
          {skill.charAt(0).toUpperCase() + skill.slice(1)}
          :
          {approach}
        </td>
      )
    })
    tableRow.push(<tr key={props.characterDetails.id + skillsPair[0]}>{items}</tr>)
    return tableRow
  }, [])

  return (
    <li className="characterSummaryTile">
      <section>
        <Link className="character-link" to={`/character/${props.characterDetails.id}`}>
          <div className="character-summary-details">
            <h3>{props.characterDetails.name}</h3>
            <p className="highConcept">{props.characterDetails.aspects.highConcept}</p>
            <p className="trouble">
              {props.characterDetails.aspects.trouble}
            </p>
          </div>
        </Link>
      </section>
      <section className="option-links">
        <Link className="character-link" to={`/character/${props.characterDetails.id}`}>
          <table className="character-skills-table">
            <caption>Approaches</caption>
            <tbody>
              { skills }
            </tbody>
          </table>
        </Link>
        <Link className="primary-button" to={`/character/${props.characterDetails.id}/edit`}>Edit</Link>
        <button className="primary-button danger-button" type="submit" onClick={handleCharacterDeletion}>Delete</button>
      </section>
    </li>
  )
}

export default CharacterSummary
