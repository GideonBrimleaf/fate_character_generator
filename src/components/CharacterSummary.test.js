import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

import CharacterSummary from './CharacterSummary';

it('renders correctly', () => {
  // Arrange
  const props = {
    characterDetails: {
      id: 1234,
      name: 'expected name',
      stunts: [],
      aspects: { expectedAspectName: 'expected aspect value' },
      approaches: { expectedApproachName: 'expected appraoch value' },
      consequences: { expectedConName: 'expected con value' },
    }
  }

  // Act
  const result = render(
    <MemoryRouter>
      <CharacterSummary {...props} />
    </MemoryRouter>
  );

  // Assert
  expect(result.asFragment()).toMatchSnapshot();
});
