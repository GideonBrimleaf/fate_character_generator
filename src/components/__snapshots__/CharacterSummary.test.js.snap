// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <li
    class="characterSummaryTile"
  >
    <section>
      <a
        class="character-link"
        href="/character/1234"
      >
        <div
          class="character-summary-details"
        >
          <h3>
            expected name
          </h3>
          <p
            class="highConcept"
          />
          <p
            class="trouble"
          />
        </div>
      </a>
    </section>
    <section
      class="option-links"
    >
      <a
        class="character-link"
        href="/character/1234"
      >
        <table
          class="character-skills-table"
        >
          <caption>
            Approaches
          </caption>
          <tbody>
            <tr>
              <td
                class="summary-skill-header"
              >
                ExpectedApproachName:expected appraoch value
              </td>
            </tr>
          </tbody>
        </table>
      </a>
      <a
        class="primary-button"
        href="/character/1234/edit"
      >
        Edit
      </a>
      <button
        class="primary-button danger-button"
        type="submit"
      >
        Delete
      </button>
    </section>
  </li>
</DocumentFragment>
`;
